### PR TITLE
Identity - Add helpful feedback page for users unable to automatically delete their Account

### PR DESCRIPTION
**Pull request by vlbee**
## What does this change?

Builds on #21783

Currently, when a user tries to delete their account and they have a paid product, the request will fail and instead notify User Help requesting a manual deletion. No feedback is provided to the user why the request failed.  

With this PR we pre-emptively advise a user what they need to do first to cancel their product before they can delete their online account, thus avoiding frustrating account deletion failures for the user and extra work for User Help.

A user is blocked from requesting an account deletion if they have a paid product, recurring contribution, or membership, and advised in each case what they need to cancel that product. 

**Implementation notes:**
No changes have been made to the [account deletion step function](https://github.com/guardian/identity-account-deletion/blob/master/cloudformation/deletion-step-function.yaml) which performs a subset of the same checks as PR #21783 to block or allow auto-account deletion. 

## What is the value of this and can you measure success?

Account deletion requests statistics are tracked by user help a make up a significant portion of their workload. This change should increase the number of users able to self-serve and decrease the  need for manual deletion by User Help. 

## Screenshots 

This shows all possible messages, which would otherwise be shown conditionally. 

image

If there is an error getting content access data from MDAPI, an error is shown to the user and the 
 error logged: 

image

If there is no reason to block a user from accessing auto-account deletion, they will be taken to the existing page to enter their password and go through the existing deletion lambda which has not changed: 
image

### CODE REVIEW

**Comment by vlbee** [comment]
@dominickendrick  Now with an error page: 
image

File: identity/app/views/profile/deletion/accountDeletionForm.scala.html [hunk 1]
```diff
@@ -0,0 +1,142 @@
+@import idapiclient.responses.Error
```
Comment (dominickendrick) [inline identity/app/views/profile/deletion/accountDeletionForm.scala.html]: To make this PR diff better you could do a `git mv <source> <destination>` and git will show that you've just renamed the file

File: identity/app/views/profile/deletion/accountDeletionBlock.scala.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+<div class="identity-wrapper monocolumn-wrapper">
+
+    @if(emailIsValidated) {
+        <div class="identity-section">
+            <p class="identity-section__text">
+                We've noticed you have an active membership, recurring contribution or subscription with the Guardian, so you won't be able to automatically delete your account online. Please follow the instructions below first and try again if you would still like delete your account.
+            </p>
+
+            </br>
```
Comment (dominickendrick) [inline identity/app/views/profile/deletion/accountDeletionBlock.scala.html]: Can we not achieve this spacing without having to resort to `<br/>`s as they are [not semantic](https://stackoverflow.com/questions/1726073/is-it-sometimes-bad-to-use-br)

**Review by dominickendrick — APPROVED** [review]
Looks good apart from suggested changes!

**Pull request merged**